plugins/in_http/http_prot.c

```diff
@@ -512,6 +513,101 @@ static ssize_t parse_payload_urlencoded(struct flb_http *ctx, flb_sds_t tag,
     return ret;
 }
 
+static ssize_t parse_payload_msgpack(struct flb_http *ctx, flb_sds_t tag,
+                                     char *payload, size_t size)
+{
+    int              ret = FLB_EVENT_ENCODER_SUCCESS;
+    struct flb_time  tm;
+    size_t           offset = 0;
+    msgpack_unpacked result;
+    msgpack_object   *record;
+    msgpack_object   *metadata;
+    msgpack_object   *data;
+    flb_sds_t        tag_from_record = NULL;
+
+
+    msgpack_unpacked_init(&result);
```

Comment:
it seems to me that we the log decoder API should be used instead, there is a mix of the old mechanism and log encoder.

Reply:
I have updated both the legacy and ng versions to use log_decoder instead of the direct msgpack API.